arm-customer-insights/2016-01-01/swagger/customer-insights.json

```diff
@@ -0,0 +1,4735 @@
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/widgetTypes/{widgetTypeName}": {
+      "get": {
+        "tags": [
+          "WidgetTypes"
+        ],
+        "operationId": "WidgetTypes_Get",
+        "x-ms-examples": { 
+          "WidgetTypes_Get": { "$ref": "../examples/WidgetTypesGet.json" } 
+        },
+        "description": "Gets a widget type in the specified hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "widgetTypeName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the widget type."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/WidgetTypeResourceFormat"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views": {
+      "get": {
+        "tags": [
+          "Views"
+        ],
+        "operationId": "Views_ListByHub",
+        "x-ms-examples": { 
+          "Views_ListByHub": { "$ref": "../examples/ViewsListByHub.json" } 
+        },
+        "description": "Gets all available views for given user in the specified hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "userId",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The user ID. Use * to retreive hub level views."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "sc
```

Comment:
There is an issue in our currently RP. For proxy resources, the field name in the properties are not camel cased. For example it's like below in the response payload:
                           {
				"properties": {
					"**ViewName**": "testView"
				},
				"type": "Microsoft.CustomerInsights/hubs/views"
			}
In swagger I name them camel cased. I tested a little with the generated C# code. It can be deserialized correctly when x-ms-client-flatten is not used. But the fields in properties will not be set correctly after using  x-ms-client-flatten. I think it's because the generated code is like below :
        [Newtonsoft.Json.JsonProperty(PropertyName = "properties.**viewName**")]
        public string ViewName { get; private set; }

properties.viewName cannot be found, only properties.ViewName exists. (I guess it can tolerate case difference only for the first character?).

Ideally we should make fix in our RP so all field names are camel cased. But it has massive impact to our service at this stage...... A lot of changes may be needed to parties have dependency on RP.

So do you have any suggestion on this? Is there anything we can do so properties.viewName can be deserialized correctly even it's named as ViewName in Http response? Thanks.